src/priority_fee.rs

```diff
@@ -193,58 +196,120 @@ impl GrpcConsumer for PriorityFeeTracker {
 
 impl PriorityFeeTracker {
     pub fn new(slot_cache_length: usize) -> Self {
+        let (sampling_txn, sampling_rxn) = channel::<(Vec<Pubkey>, bool, Option<u32>)>(1_000);
+
         let tracker = Self {
             priority_fees: Arc::new(DashMap::new()),
             slot_cache: SlotCache::new(slot_cache_length),
             compute_budget: ComputeBudget::default(),
+            sampling_sender: sampling_txn,
         };
-        tracker.poll_fees();
+        tracker.poll_fees(sampling_rxn);
         tracker
     }
 
-    fn poll_fees(&self) {
+    fn poll_fees(&self, mut sampling_rxn: Receiver<(Vec<Pubkey>, bool, Option<u32>)>) {
         let priority_fee_tracker = self.clone();
         tokio::spawn(async move {
+            let mut start = Instant::now();
             loop {
-                let global_fees =
-                    priority_fee_tracker.get_priority_fee_estimates(vec![], false, None);
-                statsd_gauge!(
-                    "min_priority_fee",
-                    global_fees.min as u64,
-                    "account" => "none"
-                );
-                statsd_gauge!("low_priority_fee", global_fees.low as u64, "account" => "none");
-                statsd_gauge!(
-                    "medium_priority_fee",
-                    global_fees.medium as u64,
-                    "account" => "none"
-                );
-                statsd_gauge!(
-                    "high_priority_fee",
-                    global_fees.high as u64,
-                    "account" => "none"
-                );
-                statsd_gauge!(
-                    "very_high_priority_fee",
-                    global_fees.very_high as u64,
-                    "account" => "none"
-                );
-                statsd_gauge!(
-                    "unsafe_max_priority_fee",
-                    global_fees.unsafe_max as u64,
-                    "account" => "none"
-                );
-                statsd_gauge!(
-                    "recommended_priority_fee",
-                    get_recommended_fee(global_fees) as u64,
-                    "account" => "none"
-                );
-                tokio::time::sleep(Duration::from_millis(1000)).await
+                let elapsed = start.elapsed().as_millis();
+                let remaining_time = 1_000u128.saturating_sub(elapsed);
```

Comment:
wont this just be 0 forever after the first second passes?

Reply:
The timeout should return err if duration will be 0 in which case the start will be reset. I did not create a large enough volume to see if thats true but without many requests that is the case…
Though I can refactor to remove timeout if that would be bettet